qt/python/mantidqt/widgets/superplot/test/SuperplotModelTest.py

```diff
@@ -0,0 +1,202 @@
+    def setUp(self):
+        pyModule = "mantidqt.widgets.superplot.SuperplotModel"
+
+        patch = mock.patch(pyModule + ".SuperplotAdsObserver")
+        self.mObs = patch.start()
+        self.mObs = self.mObs.return_value
+        self.addCleanup(patch.stop)
+
+        patch = mock.patch(pyModule + ".mtd")
+        self.mMtd = patch.start()
+        self.addCleanup(patch.stop)
+
+        self.model = SuperplotModel()
+
+    def test_init(self):
+        self.assertEqual(self.model._workspaces, [])
+        self.assertEqual(self.model._plottedData, [])
+        self.assertIsNone(self.model._plotMode)
+        self.mObs.signals.wsDeleted.connect.assert_called_once()
+        self.mObs.signals.wsRenamed.connect.assert_called_once()
+        self.mObs.signals.wsReplaced.connect.assert_called_once()
+
+    def test_addWorkspace(self):
+        self.model.addWorkspace("ws1")
+        self.assertEqual(self.model._workspaces, ["ws1"])
+        self.model.addWorkspace("ws1")
+        self.assertEqual(self.model._workspaces, ["ws1"])
+        self.model.addWorkspace("ws2")
+        self.assertEqual(self.model._workspaces, ["ws1", "ws2"])
+        self.mMtd.__getitem__.return_value = mock.Mock(spec=WorkspaceGroup)
+        self.mMtd.__getitem__.return_value.getNames.return_value = ["ws2",
+                                                                    "ws3",
+                                                                    "ws4"]
+        self.model.addWorkspace("g1")
+        self.assertEqual(self.model._workspaces, ["ws1", "ws2", "ws3", "ws4"])
+
+    def test_delWorkspace(self):
+        self.model._workspaces = ["ws1", "ws2", "ws3"]
+        self.model._plottedData = [("ws1", 1), ("ws2", 2)]
+        self.model._plotMode = self.model.BIN_MODE
+        self.model.delWorkspace("ws4")
+        self.assertEqual(self.model._workspaces, ["ws1", "ws2", "ws3"])
+        self.assertEqual(self.model._plottedData, [("ws1", 1), ("ws2", 2)])
+        self.assertEqual(self.model._plotMode, self.model.BIN_MODE)
+        self.model.delWorkspace("ws3")
+        self.assertEqual(self.model._workspaces, ["ws1", "ws2"])
+        self.assertEqual(self.model._plottedData, [("ws1", 1), ("ws2", 2)])
+        self.assertEqual(self.model._plotMode, self.model.BIN_MODE)
+        self.model.delWorkspace("ws2")
+        self.assertEqual(self.model._workspaces, ["ws1"])
+        self.assertEqual(self.model._plottedData, [("ws1", 1)])
+        self.assertEqual(self.model._plotMode, self.model.BIN_MODE)
+        self.model.delWorkspace("ws1")
+        self.assertEqual(self.model._workspaces, [])
+        self.assertEqual(self.model._plottedData, [])
+        self.assertIsNone(self.model._plotMode)
+
+    def test_getWorkspaces(self):
+        self.model._workspaces = ["ws1", "ws2", "ws3"]
+        wsList = self.model.getWorkspaces()
+        self.assertEqual(wsList, self.model._workspaces)
+        wsList = wsList[0:2]
+        self.assertEqual(wsList, ["ws1", "ws2"])
```

Comment:
could you just add a comment that this is to make sure that getWorkspace is not returing a pointer to its own list?

Reply:
done